app/models/defence_request_transitions/acknowledge.rb

```diff
@@ -18,7 +18,7 @@ def set_cco
     end
 
     def acknowledge_defence_request
-      run_transition && defence_request.save!
+      run_transition && defence_request.save! && defence_request.generate_dscc_number!
```

Comment:
How do you feel about extracting a method along the lines of `save_and_generate_dscc_number`? Even though the name hints at trouble, it feels like the defence request should handle this. What do you think?